src/test/java/com/google/sps/servlets/UpdateVendorServletTest.java

```diff
@@ -0,0 +1,169 @@
+  private static final String PARAM_IMAGE_DESCRIPTION = "altText";
+
+  // Vendor Good values
+  private static final String FIRST_NAME = "Ricardo";
+  private static final String LAST_NAME = "Chapa";
+  private static final String PHONE_NUMBER = "8118022347";
+  private static final String BLOBKEY = "image.png";
+  private static final String CURRENT_BLOBKEY = "current_blobkey";
+  private static final String IMAGE_DESCRIPTION = "mockImageDescription";
+
+  // Vendor Bad Values
+  private static final String BAD_FIRST_NAME = "R$icardo";
+  private static final String BAD_LAST_NAME = "Chap'a";
+  private static final String BAD_PHONE_NUMBER = "8118 022347";
+
+  private static HttpServletRequest mockedRequest;
+  private static HttpServletResponse mockedResponse;
+
+  private static final UpdateVendorServlet updateVendorServlet = new UpdateVendorServlet();
+
+  // Mockito instance helper
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+        new LocalDatastoreServiceTestConfig(),
+        new LocalUserServiceTestConfig(), 
+        new LocalBlobstoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    mockedRequest = mock(HttpServletRequest.class);
+    mockedResponse = mock(HttpServletResponse.class);
+    when(mockedRequest.getParameter(PARAM_FIRST_NAME)).thenReturn(FIRST_NAME);
+    when(mockedRequest.getParameter(PARAM_LAST_NAME)).thenReturn(LAST_NAME);
+    when(mockedRequest.getParameter(PARAM_PHONE_NUMBER)).thenReturn(PHONE_NUMBER);
+    when(mockedRequest.getParameter(PARAM_CURRENT_BLOBKEY)).thenReturn(BLOBKEY);
+    when(mockedRequest.getParameter(PARAM_CURRENT_BLOBFILE)).thenReturn(CURRENT_BLOBKEY);
+    when(mockedRequest.getParameter(PARAM_IMAGE_DESCRIPTION)).thenReturn(IMAGE_DESCRIPTION);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  // TODO Find a way to mock blobstore service and add its Tests
+  // Actual: Without try block gives the following error
+  // java.lang.IllegalStateException: Must be called from a blob upload callback request. 
+  @Test
+  public void hasBadParameters() throws IOException {
+    try {
+    helper.setEnvIsLoggedIn(true);
+    when(mockedRequest.getParameter(PARAM_FIRST_NAME)).thenReturn(BAD_FIRST_NAME);
+
+    updateVendorServlet.doPost(mockedRequest, mockedResponse);
+
+    verifyBadRequest(mockedResponse);
+    } catch (IllegalStateException e) {
+      System.out.println(e);
+    }
+  }
+
+  @Test
+  public void userIsNotLoggedIn() throws IOException {
+    helper.setEnvIsLoggedIn(false);
+
+    updateVendorServlet.doPost(mockedRequest, mockedResponse);
+    verifyBadAccess(mockedResponse);
+  }
+
+  @Test
+  public void allBadParameters() throws IOException {
+    final boolean expected = true;
+    final boolean actual = 
+        updateVendorServlet.badInputs(BAD_FIRST_NAME, BAD_LAST_NAME, BAD_PHONE_NUMBER);
+    Assert.assertEquals(expected, actual);
```

Comment:
Nit: is there an assertTrue function? Also, you can just delete the "expected" variable and use the expected value directly, like below:

Assert.assertEquals(true, actual);
Assert.assertEquals(false, actual);